Tests/LibCore/TestLibCoreMimeType.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2025, Andreas Kling <andreas@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "TestLibCoreMimeType.h"
+#include <AK/Vector.h>
+#include <LibCore/MimeData.h>
+#include <LibTest/TestCase.h>
+
+
+void check_filename_mimetype(AK::Vector<StringView> const& filepaths, StringView const& expected_mimeType)
+{
+    for (auto const& filename : filepaths)
+    {
+        dbgln(filename, "\n");
+        auto const& guessed_mime_type = Core::guess_mime_type_based_on_filename(filename);
+        EXPECT_EQ(guessed_mime_type, expected_mimeType);
+    }
+}
+
+auto text_plain_filenames = Vector {"main.c"sv, "hello.txt"sv, ".history"sv, ".shellrc"sv, "CMakeList.txt"sv,};
+//FIXME: fails because .xht extension is in MimeType text/html and application/xhtml+xml
```

Comment:
It's kind of hard to find an authoritative reference for .xht files' preferred essence. But surely application/xhtml+xml is much more correct than text/html. A patch to fix up this duplication would be a good follow-up.